server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java

```diff
@@ -336,8 +332,6 @@ public void testNoExtensionsFile() throws Exception {
     }
 
     public void testEmptyExtensionsFile() throws Exception {
-        Path extensionDir = createTempDir();
```

Comment:
This one needs to stay (although use a different variable name to avoid confusion).  This tests an empty YAML file so it can't use the same one as all the other tests.  This one will overwrite the others so depending on the order the tests are executed in (which I think is nondeterministic) it could cause random failures unless this test is run last.